tools/js-optimizer.js

```diff
@@ -7911,6 +7911,26 @@ function asmLastOpts(ast) {
   });
 }
 
+// Contrary to the name this does not eliminate actual dead functions, only
+// those marked as such with DEAD_FUNCTIONS
+function eliminateDeadFuncs(ast) {
+  assert(asm);
+  assert(extraInfo && extraInfo.dead_functions);
+  var dead_functions = extraInfo.dead_functions;
+  traverseGeneratedFunctions(ast, function (fun, type) {
+    for (var i = 0; i < dead_functions.length; i++) {
```

Comment:
instead of linear traversal use a `set()`, and check `func[1] in thatSet`. see other examples in the file